javalib/src/main/scala/java/net/HttpCookie.scala

```diff
@@ -0,0 +1,374 @@
+            )) {
+          cookie.setSecure(true)
+        } else if (propName.regionMatches(
+              true,
+              0,
+              HttpOnly,
+              0,
+              HttpOnly.length()
+            )) {
+          cookie.setHttpOnly(true)
+        } else if (propName.regionMatches(
+              true,
+              0,
+              Version,
+              0,
+              Version.length()
+            )) {
+          try {
+            cookie.setVersion(java.lang.Integer.valueOf(propValue))
+          } catch {
+            case _: NumberFormatException =>
+              throw new IllegalArgumentException(
+                "Illegal cookie max-age attribute"
+              )
+          }
+        }
+
+        i += 1
+
+      }
+
+      cookie
+
+    }
+
+  }
+
+  // NAME = attr
+  // attr = token
+  // token = 1*<any CHAR except CTLs or separators>
+  // separators = "(" | ")" | "<" | ">" | "@"
+  //                | "," | ";" | ":" | "\" | <">
+  //                | "/" | "[" | "]" | "?" | "="
+  //                | "{" | "}" | SP | HT
+  // CHAR = <any US-ASCII character (octets 0 - 127)>
+  // CTL = <any US-ASCII control character (octets 0 - 31) and DEL (127)>
+  private def validateName(name: String): String = {
+    if (name.length == 0)
+      throw new IllegalArgumentException("Illegal cookie name")
+
+    var i = 0
+    while (i < name.length()) {
+      val char = name.charAt(i)
+      if (char < 32 || // CTL
+          char == 127 || // CTL
+          char == '(' ||
+          char == ')' ||
+          char == '<' ||
+          char == '>' ||
+          char == '@' ||
+          char == ',' ||
+          char == ';' ||
+          char == ':' ||
+          char == '\\' ||
+          char == '"' ||
+          char == '/' ||
+          char == '[' ||
+          char == ']' ||
+          char == '?' ||
+          char == '=' ||
+          char == '{' ||
+          char == '}' ||
+          char == ' ' ||
+          char == '\t') {
+        throw new IllegalArgumentException("Illegal cookie name")
+      }
+
+      i += 1
+    }
+
+    name
+
+  }
+
+}
+
+final class HttpCookie private (
+    private var _comment: String = null,
+    private var _commentURL: String = null,
+    private var _httpOnly: Boolean = false,
+    private var _discard: Boolean = false,
+    private var _domain: String = null,
+    private var _maxAge: Long = -1,
+    private var _name: String = null,
+    private var _path: String = null,
+    private var _portlist: String = null,
+    private var _secure: Boolean = false,
+    private var _value: String = null,
+    private var _version: Int = 1
+) extends Cloneable {
+
+  def this(name: String, value: String) = {
+    this(
+      _name =
+        (if (name.length() == 0) {
+           throw new IllegalArgumentException("Illegal cookie name")
+         } else
+           HttpCookie.validateName(name)),
+      _value = value
+    )
+  }
+
+  override def equals(obj: Any): Boolean = {
```

Comment:
A style point.  I recommend moving the `override` of `hashCode` below
to be either just before or just after the override of `equals()`.
That will make it easier for people reading the code who go on alert
when they see `equals` being overridden to convince themselves 
that `hashCode` is properly overridden in co-ordination.

As long as one thing is moving, I suggest moving `toString()` to
the same block so that overrides are together and easier to find.

Reply:
Late at night now. By the light of day I will try to post a URL
to the current "Best Practice" for an `equals` method.

It helped me last week when I had to write an `equals` for 
`fileKey`s.  Ultimately, because of some wierdness with `fileKey`,
I could not follow it exactly, but at least I knew what I was deviating from and why.

Reply:
Moved things around and implemented a new `equals` as per the advice you linked